app/models/text_field_type.rb

```diff
@@ -25,14 +27,24 @@ def mapping_field_name
   end
 
   def text_present
-    errors.add(:text, "must be present") if @text.empty?
+    errors.add(:text, 'must be present') if @text.empty?
   end
 
   def text_length
     validator = LengthValidator.new(validations[:length].merge(attributes: [:text]))
     validator.validate_each(self, :text, text)
   end
 
+  def text_unique
+    unless Field.find_by_name(field_info.name).field_items.jsonb_contains(:data, text: text).empty?
```

Comment:
`Field` `name`s are not globally unique - there could be another `Field` with the same name in a different `ContentType`, which would break this validation. If you wanted to use the structure you have in place here, you'd want to scope to the `ContentType` first. However, I can suggest something even better - why are we looking up the `Field` twice (once in Cortex, and once in the `FieldType`)? Either pass the entire `Field` object (since we have access to it) to `FieldType`, or if for some reason you really want to perform the extra lookup, at the very least pass its `ID`. This way, you don't have to guess about its `ContentType`.